src/statuscake-monitor.go

```diff
@@ -0,0 +1,286 @@
+	url      string
+	username string
+	client   *http.Client
+}
+
+// AnnotationInfo is the annotation information structure for AnnotationMap
+type AnnotationInfo struct {
+	name     string
+	dataType string
+}
+
+// AnnotationMap holds all the enabled annotations for StatusCake
+var AnnotationMap = map[string]AnnotationInfo{
+	"monitor.stakater.com/statuscake-check-rate":      AnnotationInfo{"CheckRate", "int"},       // Int (0-24000)
+	"monitor.stakater.com/statuscake-test-type":       AnnotationInfo{"TestType", "string"},     // String (HTTP, TCP, PING)
+	"monitor.stakater.com/statuscake-paused":          AnnotationInfo{"Paused", "bool"},         // Int (0,1)
+	"monitor.stakater.com/statuscake-ping-url":        AnnotationInfo{"PingURL", "string"},      // String (url)
+	"monitor.stakater.com/statuscake-follow-redirect": AnnotationInfo{"FollowRedirect", "bool"}, // Int (0,1)
+	"monitor.stakater.com/statuscake-port":            AnnotationInfo{"Port", "int"},            // Int (TCP Port)
+	"monitor.stakater.com/statuscake-trigger-rate":    AnnotationInfo{"TriggerRate", "int"},     // Int (0-60)
+	"monitor.stakater.com/statuscake-contact-group":   AnnotationInfo{"ContactGroup", "string"}} // String (0-60)
+
+// buildUpsertForm function is used to create the form needed to Add or update a monitor
+func buildUpsertForm(m Monitor) url.Values {
+	f := url.Values{}
+	f.Add("WebsiteName", m.name)
+	f.Add("WebsiteURL", m.url)
+	if val, ok := m.annotations["monitor.stakater.com/statuscake/check-rate"]; ok {
+		f.Add("CheckRate", val)
+		delete(m.annotations, "monitor.stakater.com/statuscake/check-rate")
+	} else {
+		f.Add("CheckRate", "300")
+	}
+	if val, ok := m.annotations["monitor.stakater.com/statuscake/test-type"]; ok {
+		f.Add("TestType", val)
+		delete(m.annotations, "monitor.stakater.com/statuscake/test-type")
+	} else {
+		f.Add("TestType", "HTTP")
+	}
+	for key, value := range m.annotations {
+		if (AnnotationInfo{}) != AnnotationMap[key] {
+			meta := AnnotationMap[key]
+			switch strings.ToLower(meta.dataType) {
+			case "int":
+				f.Add(meta.name, value)
+			case "string":
+				f.Add(meta.name, value)
+			case "bool":
+				value = strings.ToLower(value)
+				if value == "true" {
+					f.Add(meta.name, "1")
+				} else {
+					f.Add(meta.name, "0")
+				}
+			}
+		}
+	}
+	return f
+}
+
+// Setup function is used to initialise the StatusCake service
+func (service *StatusCakeMonitorService) Setup(p Provider) {
+	service.apiKey = p.ApiKey
+	service.url = p.ApiURL
+	service.username = p.Username
+	service.client = &http.Client{}
+}
+
+// GetByName function will Get a monitor by it's name
+func (service *StatusCakeMonitorService) GetByName(name string) (*Monitor, error) {
+	monitors := service.GetAll()
+	for _, monitor := range monitors {
+		if monitor.name == name {
+			return &monitor, nil
+		}
+	}
+	errorString := "GetByName Request failed for name: " + name
+	return nil, errors.New(errorString)
+	// u, err := url.Parse(service.url)
```

Comment:
@Anish-Gupta-Bose wondering what is need to keep these commented lines? Please remove them.